src/app.rs

```diff
@@ -90,15 +86,25 @@ impl ReceiptHandler {
         if self.receipt_set.is_empty() {
             return false;
         }
+
+        // Get any key
         let uuid = *self.receipt_set.keys().next().unwrap();
-        let mut queue = self.receipt_set.remove(&uuid).unwrap();
-        if let Some((timestamps, receipt)) = queue.get_data() {
-            signal_manager.send_receipt(uuid, timestamps, receipt);
-            return true;
-        }
-        if !queue.is_empty() {
-            self.receipt_set.insert(uuid, queue);
-        }
+
+        let j = self.receipt_set.entry(uuid);
+        match j {
+            Entry::Occupied(mut e) => {
+                let u = e.get_mut();
+                if let Some((timestamps, receipt)) = u.get_data() {
+                    signal_manager.send_receipt(uuid, timestamps, receipt);
+                    if u.is_empty() {
+                        self.receipt_set.remove_entry(&uuid);
+                    }
+                    return true;
+                }
+            }
+            Entry::Vacant(_) => {}
+        };
+        self.receipt_set.remove_entry(&uuid);
```

Comment:
There is no need for this call, because this can happen only in the vacant case.

Reply:
Maybe it even makes sense to move the `false` return value up here, and replace the `return true` in line 105 by `true`.